index.bs

```diff
@@ -2334,6 +2309,121 @@ The [=default allowlist=] for this feature is <code>["self"]</code>.
 
 Note: If the document's origin is not allowed to use the <code>"xr-spatial-tracking"</code> feature policy any [=immersive sessions=] will be blocked, because all [=immersive sessions=] require some use of spatial tracking. [=Inline sessions=] will still be allowed, but restricted to only using the {{XRReferenceSpaceType/"viewer"}} {{XRReferenceSpace}}.
 
+
+Permissions API Integration {#permissions}
+---------
+
+The [[!permissions]] API provides a uniform way for websites to request permissions from users and query which permissions they have been granted. 
+
+The <dfn enum-value for="PermissionName">"xr"</dfn> [=powerful feature=]’s permission-related algorithms and types are defined as follows: 
+
+
+<dl>
+<dt>[=permission descriptor type=]</dt>
+<dd>
+
+<pre class="idl">
+dictionary XRPermissionDescriptor: PermissionDescriptor {
+  XRSession session;
```

Comment:
It seems odd to me that `XRPermissionDescriptor` would contain an `XRSession` as this is what developers would need to call `permissions.query()` and ideally, they may want to do that before calling `requestSession()`. However, `requestSession()` is what provides the session and querying about the permission status at that point becomes moot.

Reply:
It's because there's no user-exposed handle for the "device" except for the session itself. The permissions integration needs to check against a device, and also needs to be able to deal with the device being different -- e.g. if you plug in a device that supports bounded-floor spaces, you want the permissions check to work.

We could replace this with a mode enum, and use the "current XR device" instead, setting the list of granted features on the device instead of the session.

Reply:
Why does the permission need to be checked against a device?

Reply:
Some devices don't support a feature, period.This is the existing behavior of XR features, I'm wary about changing how this works without broader agreement.